src/main/java/org/wildfly/security/mechanism/MechanismUtil.java

```diff
@@ -76,7 +77,7 @@ private MechanismUtil() {}
             CredentialCallback credentialCallback = new CredentialCallback(PasswordCredential.class, passwordAlgorithm, matchParameters);
 
             try {
-                MechanismUtil.handleCallbacks(passwordAlgorithm, callbackHandler, credentialCallback);
+                MechanismUtil.handleCallbacks(log, callbackHandler, credentialCallback);
```

Comment:
+1 This looks weird, but it is correct fix.
(first param should be mechanism to prefix logging, passwordAlgorithm should not be here)